src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -38,12 +43,16 @@
     private final Logger logger = LoggerFactory.getLogger("bio.terra.service.filedata.DrsService");
 
     private static final String DRS_OBJECT_VERSION = "0";
+    // atomic counter that we incr on request arrival and decr on request response
+    private static int currentDRSRequests = 0;
```

Comment:
This isn't atomic. There are classes that you can use to get atomic behavior across competing threads.
[AtomicInteger](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/atomic/AtomicInteger.html)